### PR TITLE
Adding method in API to get linked entities version

### PR DESCRIPTION
**Pull request by Mun1z**
**Depends on #114.**

Adding method in ML-Git API, which aims to list all linked entities of a specific entity versions. This method uses the Github API.

```
def get_linked_entities(self, entity_name, entity_version, metadata_repo_name):
    """Get a list of linked entities found for an entity version.

    Args:
        entity_name (str): The name of the entity you want to get the linked entities.
        entity_version (str): The version of the entity you want to get the linked entities.
        metadata_repo_name (str): The repository name where the metadata is located in GitHub.

    Returns:
        list of LinkedEntity.
    """
```

**Usage example:** 
```
from ml_git import api
github_token = ''
api_url = 'https://api.github.com'
manager = api.init_entity_manager(github_token, api_url)

linked_entities = manager.get_linked_entities('entity_name', 'entity_version', metadata_repo_name='user/metadata_repository')
```

The variable `linked_entities ` should look something like:
```
[{
  "tag": "test__datasets-ex__1",
  "name": "datasets-ex",
  "version": "1",
  "entity_type": "dataset"
}, {
  "tag": "test__labels-ex__1",
  "name": "labels-ex",
  "version": "1",
  "entity_type": "labels"
}]
```

### CODE REVIEW

**Review by lucasncabral — APPROVED** [review]

File: ml_git/relationship/models/metadata.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+"""
+© Copyright 2021 HP Development Company, L.P.
+SPDX-License-Identifier: GPL-2.0-only
+"""
+import json
+
+
+class Metadata:
```
[the remark this answers is not in the document]
Comment (Mun1z) [reply ml_git/relationship/models/metadata.py:8]: In the entity's output it was defined that it should be "metadata".

**Review by user-mlgit — APPROVED** [review]

**Review by gilMars — APPROVED** [review]

**Review by andersonpablosilva — APPROVED** [review]

File: ml_git/relationship/models/linked_entity.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+"""
+© Copyright 2021 HP Development Company, L.P.
+SPDX-License-Identifier: GPL-2.0-only
+"""
+import json
+
+
+class LinkedEntity:
+    """Class that's represents an linked ml-entity.
```
[the remark this answers is not in the document]
Comment (lucasncabral) [reply ml_git/relationship/models/linked_entity.py]: Text has been updated in 0ad9e12.

File: docs/api/README.md [hunk 3]
```diff
@@ -360,6 +371,23 @@ class Storage:
...
+
+```python
+class LinkedEntity:
+    """Class that represents a linked ml-entity.
+
+    Attributes:
+        name (str): The name of the entity.
+        entity_type (str): The type of the ml-entity (datasets, models, labels).
+        version (str): The version of the ml-entity.
+        tag (str): The tag of the ml-entity spec version.
```
Comment (gpretto) [inline docs/api/README.md:386]: Is it necessary use entity for the 'entity_type'? I would suggest have all attributes containing the name entity or neither.

File: ml_git/relationship/models/linked_entity.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class LinkedEntity:
+    """Class that represents a linked ml-entity.
+
+    Attributes:
+        name (str): The name of the entity.
+        entity_type (str): The type of the ml-entity (datasets, models, labels).
+        version (str): The version of the ml-entity.
+        tag (str): The tag of the ml-entity spec version.
```
Comment (gpretto) [inline ml_git/relationship/models/linked_entity.py:15]: Is it necessary use entity for the 'entity_type'? I would suggest have all attributes containing the name entity or neither.

**Review by gpretto — CHANGES_REQUESTED** [review]
Few changes.

File: ml_git/relationship/models/linked_entity.py (hunk 4 above)
Comment (Mun1z) [reply ml_git/relationship/models/linked_entity.py]: Done on commit: https://github.com/HPInc/ml-git/pull/115/commits/8e642e1fed659efb88308ccccd96a90178c24697

File: docs/api/README.md (hunk 3 above)
Comment (Mun1z) [reply docs/api/README.md:404]: Done on commit: https://github.com/HPInc/ml-git/pull/115/commits/8e642e1fed659efb88308ccccd96a90178c24697

File: docs/api/README.md [hunk 5]
```diff
@@ -360,6 +372,23 @@ class Storage:
...
+<summary><code> LinkedEntity </code></summary>
+<br>
+
+```python
+class LinkedEntity:
+    """Class that represents a linked ml-entity.
+
+    Attributes:
+        name (str): The name of the entity.
+        entity_type (str): The type of the ml-entity (datasets, models, labels).
```
Comment (gpretto) [inline docs/api/README.md:385]:
```suggestion
        type (str): The type of the ml-entity (datasets, models, labels).
```

I think we lost this change.

File: ml_git/relationship/entity_manager.py [hunk 6]
```diff
@@ -137,3 +137,29 @@ def get_entity_versions(self, entity_name, metadata_repo_name):
             entity_version = SpecVersion(spec_tag_yaml)
             versions.append(entity_version)
         return versions
+
+    def get_linked_entities(self, entity_name, entity_version, metadata_repo_name):
+        """Get a list of linked entities found for an entity version.
+
+        Args:
+            entity_name (str): The name of the entity you want to get the linked entities.
+            entity_version (str): The version of the entity you want to get the linked entities.
```
Comment (gpretto) [inline ml_git/relationship/entity_manager.py:146]:
```suggestion
            name (str): The name of the entity you want to get the linked entities.
            version (str): The version of the entity you want to get the linked entities.
```
In other parts of this code we use the variable name without the entity. Unless it is not valid for this scenario, I think we can keep the same pattern.

**Review by gpretto — CHANGES_REQUESTED** [review]

File: ml_git/relationship/entity_manager.py (hunk 6 above)
Comment (Mun1z) [reply ml_git/relationship/entity_manager.py]: Yes, done on commit: https://github.com/HPInc/ml-git/pull/115/commits/d6f30e8e5d15e7e96d06733c9182e0872c051978

File: docs/api/README.md (hunk 5 above)
Comment (Mun1z) [reply docs/api/README.md]: Nice, done on commit: https://github.com/HPInc/ml-git/pull/115/commits/d6f30e8e5d15e7e96d06733c9182e0872c051978

**Review by gpretto — APPROVED** [review]

**Pull request merged**